meta-avocado/recipes-avocado/sdk/avocado-sdk-metadata.bb

```diff
@@ -0,0 +1,194 @@
+    os.makedirs(deploy_dir_rpm, exist_ok=True)
+
+    # Define file paths
+    repo_file_path = os.path.join(d_dir, 'avocado-sdk.repo')
+    map_file_path = os.path.join(deploy_dir_rpm, 'avocado-repo.map')
+
+    # Combine architectures into a unique set
+    all_archs = set(pkg_archs + sdk_pkg_archs)
+
+    priority = 1
+
+    # Overwrite map file and repo file initially
+    with open(map_file_path, 'w') as map_f:
+        map_f.write('')
+    with open(repo_file_path, 'w') as repo_f:
+        repo_f.write('')
+
+    # --- Unconditionally add the mapping for this recipe's own arch ---
+    # Note: This specific logic remains outside the loop as it's unconditional
+    # and slightly different (doesn't add to repo file).
+    map_value_path_for_all = f"{distro_codename}/sdk/all"
+    with open(map_file_path, 'a') as map_f:
+        bb.note(f"Adding unconditional map entry: all_avocadosdk={map_value_path_for_all}")
+        map_f.write(f"all_avocadosdk={map_value_path_for_all}\n")
+
+    # Append to files for other architectures found
+    with open(map_file_path, 'a') as map_f, open(repo_file_path, 'a') as repo_f:
+        for arch in sorted(list(all_archs)): # Sort for consistent output order
+            # Skip only dummy architectures
+            if arch.startswith('sdk-provides-dummy'):
+                continue
+
+            # Skip the special arch handled unconditionally above
+            if arch == "all_avocadosdk":
+                 continue
+
+            # arch_dir calculated for directory check and potential map key
+            arch_dir = arch.replace('-', '_')
+
+            # --- For all other arches, check directory FIRST ---
+            check_dir = os.path.join(deploy_dir_rpm, arch_dir)
+            if os.path.isdir(check_dir):
+                # --- Determine Paths and Names ---
+                repo_details = _determine_repo_paths(
+                    arch, arch_dir, distro_codename, machine_short_name, sdk_suffix_us, machine, bb
+                )
+
+                # --- Write to Map File ---
+                # Note: We still call write_map_entry even for all_avocadosdk case inside _determine_repo_paths,
+                # but it won't write to the repo file.
+                _write_map_entry(
+                    map_f, arch_dir, repo_details["map_value_path"], bb
+                )
+
+                # --- Write to Repo File ---
+                # Pass arch to helper for accurate warnings
+                if _write_repo_entry(
+                    repo_f, base_repo_url,
+                    repo_details["repo_url_path"],
+                    repo_details["repo_name"],
+                    repo_details["repo_section_name"],
+                    gpg_check, priority, arch, bb
+                ):
+                    priority += 1 # Increment priority only if repo entry was written
+            else:
+                bb.note(f"Skipping arch '{arch}' as directory '{check_dir}' does not exist")
```

Comment:
Extract the body of this `for` loop into a helper function that only takes `arch` as an argument. Python has lexical scope so you can capture variables defined in the outer scope (e.g. `bb`, `deploy_dir_rpm`, `base_repo_url`, `distro_codename`, `machine`, `sdk_suffix_us`, `gpg_check`, `machine_short_name`, `repo_file_path`, `priority`) automatically by name inside of an inner function. Lambda capture make sense here since these values are all constant. The same technique can be applied to the other inner helper functions like `determine_repo_paths` and `write_repo_entry`.

Reply:
Updated the function and loop